### ISSUE TITLE
[suggestion] A Discussions page for source-code specific questions, discussions  ? 

### ISSUE THREAD

**Issue by victorel-petrovich** [notepad-plus-plus/notepad-plus-plus#13958]
I've been struggling over the last couple of days attempting to work on a future PR on an issue myself. But I'm totally inexperienced, I have some questions about the source code and I don't know where or whom to ask them.

The community forum seems more for asking/discussing Notepad++ as an app at user-level, not the coding of it.
The Issues here are for ... you know what; not for a question about how a piced of  code works, in any case. 
And it's not a general C++ question that would be suitable for other general help websites. 

So that's why the suggestion.
Here on github, the maintainers could open the "Discussion" tab, where they could create a custom Category just for this purpose. All other default junk categories could be deleted, to keep it simple (but in future can be easily changed into a Section, with several Categories inside, if will be really needed). 

Then one could ask there a question on source code, and who wants & has time will answer, without to have to bother someone in particular.

@donho , @dail8859 (as I assume you're the main maintainers here) , what do you think? 

I've actually just experimented with this idea, here's the link and screenshots:
https://github.com/victorel-petrovich/notepad-plus-plus_lastRecentFileList.cpp/discussions

![look_for_users](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/11040279/6c59aede-a7b4-4731-ba31-26d8987efad2)
![how_create_category](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/11040279/48e938ba-9f0a-40ff-86fc-07f96332e57a)
![editCategory](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/11040279/5752aec3-912f-4b15-88bb-9d95f3601a32)

**Comment by alankilborn** [comment]
@victorel-petrovich 

I think the general idea here is a pretty one.

**Comment by victorel-petrovich** [comment]
@alankilborn  but it looks like Npp's authors didn't notice this suggestion or don't think much of it. 

The alternative could be to have a separate section at https://community.notepad-plus-plus.org/ for this, although here I think would be a bit better as the source-code is "in hand's reach", can be referenced easier.  What do you think? 
A post to discuss this in the community?

**Comment by victorel-petrovich** [comment]
ok

**Comment by donho** [comment]
Use https://community.notepad-plus-plus.org/category/2/general-discussion for any dev discussion,

**Comment by alankilborn** [comment]
IMO, not an ideal place for that.  If so, why have ANY categories on that site?

**Comment by pryrt** [comment]
@donho, When Alan and I were chatting earlier this week, we thought maybe this N++\-code specific discussion could be merged with the Plugin Development into one "Technical" category.  I thought maybe a new name of "Technical Details" or "Development Details", with a description of "Deep, technical discussion about the development of Plugins and Notepad++ source code".

I know this _seems_ to eliminate much of the "**Plugin Development**: All the issues (questions/bug reports/feature requests) about plugins".  But really, the development details of Plugins are what was the bulk of that Category.  Questions about _using_ plugins often end up in Help Wanted. And the individual plugin repos are responsible for their own issue and feature-request tracking.

If you're okay with letting us change the focus of the **Plugin Development** Category, I could make the name and description change in the Forum.

**Comment by donho** [comment]
@pryrt 
OK for changing "Plugin Development" into "Notepad++ source code discussion" or somethings-else (if you find a better name).
My intention is to separate project and discussion. 
Have you enough right to rename it?

**Comment by victorel-petrovich** [comment]
@pryrt 
> with a description of "Deep, technical discussion about the development of Plugins and Notepad++ source code".

I think better to remove "Deep", not to intimidate more beginner programmers/contributors who have a question.

**Comment by alankilborn** [comment]
@pryrt 

Whatever it is named, I think it should attempt to cover these types of questions (at least, maybe I've forgotten something):

* specific source code questions (not general stuff, e.g. "describe to me how the software is architected, and what is found where")

* build-specific questions (for those that are trying to build from source but are stuck on something)

* plugin development (as it always has, BUT maybe the title of it -- whatever that ends up being -- doesn't make it sound like this is the primary reason for the category's existence)

**Comment by pryrt** [comment]
@donho,
> Have you enough right to rename it?

Yes.

@victorel-petrovich ,
> I think better to remove "Deep"

Makes sense

> Then perhaps in a pinned topic there, can stress that the questions should be specific to Npp or it's plugins

I agree completely

@alankilborn ,
>  I think it should attempt to cover these types of questions

I used your bullets (slightly edited) in the pinned topic.

> Whatever it is named

I chose [Notepad++ & Plugin Development](https://community.notepad-plus-plus.org/category/5/notepad-plugin-development) with description "Technical discussion of building or contributing to Notepad++ or Plugin codebases" -- my earlier name suggestions were too generic, so I hope this captures the focus without being too wordy.

**Comment by victorel-petrovich** [comment]
@pryrt  thank you,
I just read https://community.notepad-plus-plus.org/topic/24812/read-this-first , sounds pretty good!

However, what's inside the last part, grey area, IMO is rather off topic to the category. 
If it's a feature request or bug for a specific, existing plugin, then definitely it should be posted at the respective plugin's github (or other, if the case) page. 
If it's a proposal for a new plugin, or new feature that the author doesn't know which plugin could incorporate it (if any), then ... I'm not so sure. 
But it's probably best to keep the focus of the category on *how-to* on code/development, rather than feature proposals.

**Comment by alankilborn** [comment]
I think the "grey area" is really saying "don't do the following" in Peter's typical nice-guy style. :-)
